source/Armament.cpp

```diff
@@ -26,17 +26,29 @@ using namespace std;
 
 
 // Add a gun hardpoint (fixed-direction weapon).
-void Armament::AddGunPort(const Point &point, const Angle &angle, bool isParallel, bool isUnder, const Outfit *outfit)
+void Armament::AddGunPort(const Point &point, const Angle &angle, bool isParallel, bool isUnder, const Outfit *outfit, bool isEnabled, bool isFrugal, bool hasIndividualFrugality)
 {
-	hardpoints.emplace_back(point, angle, false, isParallel, isUnder, outfit);
+	if(outfit == nullptr)
+	{
+		hasIndividualFrugality = false;
+		isEnabled = true;
+		isFrugal = false;
+	}
```

Comment:
This shouldn't be necessary because these values should come through from the calls in Ship::FinishLoading and Ship::Load  if `outfit`actually is a null.  Not necessarily sure how to test if it is needed, however.